src/_provider/singleAbstract.ts

```diff
@@ -94,6 +101,60 @@ export abstract class SingleAbstract {
     return this._getStatus();
   }
 
+  abstract _setStartDate(startDate: definitions.startFinishDate): void;
+
+  public setStartDate(startDate: definitions.startFinishDate): SingleAbstract {
+    if (this.supportsDates()) {
+      this._setStartDate(startDate);
+    }
+    return this;
+  }
+
+  abstract _getStartDate(): definitions.startFinishDate | null;
+
+  public getStartDate(): definitions.startFinishDate | null {
+    if (this.supportsDates()) {
+      return this._getStartDate();
+    }
+    return null;
+  }
+
+  abstract _setFinishDate(finishDate: definitions.startFinishDate): void;
+
+  public setFinishDate(finishDate: definitions.startFinishDate): SingleAbstract {
```

Comment:
Same here could be null

Reply:
Resolved by adjusting `definitions.startFinishDate` to be `string | null` and removing the other `| null`